src/zope/app/apidoc/apidoc.py

```diff
@@ -39,21 +38,31 @@ def __init__(self, parent, name):
         self.__parent__ = parent
         self.__name__ = name
 
+    def __locate(self, obj, name):
+        # In general, *always* doing this is not just weird (threads), it also leads to
+        # circular __parent__ chains, which causes issues with things like
+        # zope.securitypolicy
+        # (https://github.com/zopefoundation/zope.securitypolicy/issues/8), and
+        # it can lead to bad behaviour with sharing persistent objects
+        # across ZODB connections (which may close). Fortunately, our traverser
+        # takes care of this by making sure we are always located at an (equivalent)
+        # fresh root.
+        locate(obj, self, name)
+        return obj
+
     def get(self, key, default=None):
         """See zope.container.interfaces.IReadContainer"""
         utility = zope.component.queryUtility(IDocumentationModule, key, default)
-        if utility != default:
-            locate(utility, self, key)
+        if utility is not default:
+            utility = self.__locate(utility, key)
```

Comment:
The tests are broken (in that we can't spider all the links without running into the securitypolicy recursion issue) unless we do *something*, so it does need to be solved IMO. 

I like the copy approach because (1) it's easy and (2) it doesn't involve hardcoding anything or making assumptions about where ++apidoc++ is located. (On my own site i'm thinking of using this in, we have several sub-sites with their own component registries, both persistent and of the `z3c.baseregistry` type, so I don't like the thought of hardcoding paths.)